src/packageSourceGenerator/PackageSourceGenerator.proj

```diff
@@ -57,6 +44,23 @@
            Condition="('$(PackageName)' == '' or '$(PackageVersion)' == '') and '$(PackageCSV)' == ''" />
   </Target>
 
+  <!-- Parse the passed in CSV file into PackageInfo items with metadata. -->
+  <UsingTask TaskName="ParsePackagesCsv" AssemblyFile="$(PackageSourceGeneratorTaskAssembly)" />
+  <Target Name="ParseCsv"
+          Condition="'$(PackageCSV)' != ''"
+          AfterTargets="ValidateRestoreCsvContentProperties">
+    <ParsePackagesCsv CsvPath="$(PackageCSV)">
+      <Output TaskParameter="PackageInfo" ItemName="PackageInfo" />
+      <Output TaskParameter="GroupedPackageInfo" ItemName="GroupedPackageInfo" />
+    </ParsePackagesCsv>
+
+    <ItemGroup>
+      <PackageDownload Include="@(GroupedPackageInfo->Metadata('PackageName'))"
+              Version="%(PackageVersions)"
+              Condition="'@(GroupedPackageInfo)' != ''" />
+    </ItemGroup>
```

Comment:
`PackageDownload` and `PackageReference` items which are read by NuGet need to either be defined statically at evaluation time or inside a target that runs before `CollectPackageReferences` / `CollectPackageDownloads`.

So I would add a `BeforeTargets="CollectPackageDownloads"` to this target.

Reply:
The problem is that CollectPackageDownloads runs before the referenced PackageSourceGeneratorTask.csproj gets built as it is defined as a ProjectReference.  I am not sure what the best way to trigger the building of PackageSourceGeneratorTask.csproj.  Changing AfterTargets="ValidateRestoreCsvContentProperties" to BeforeTargets="CollectPackageDownloads" doesn't solve this problem.

Reply:
Oh I see. While I usually don't recommend doing that, you could use an inline C# task.

Reply:
Ok, that thought crossed my mind but didn't think it would be acceptable.  I don't want to spend much time on this and would just like a working solution so I will proceed with this.  We can always refactor later.  Thanks for the feedback.